plugins/inputs/ecs/types_test.go

```diff
@@ -0,0 +1,61 @@
+package ecs
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_parseTask(t *testing.T) {
+	r, err := os.Open("testdata/metadata.golden")
+	if err != nil {
+		t.Errorf("error opening test files")
+	}
+	parsed, err := unmarshalTask(r)
+	if err != nil {
+		t.Errorf("error parsing task %v", err)
+	}
+	assert.Equal(t, validMeta, *parsed, "Got = %v, want = %v", parsed, validMeta)
```

Comment:
No need to add a custom message, its just too much work to write and maintain (especially for those of us with a programming induced RSI).  Testify will do something very similar by default.  I also use `require` for almost all things instead of `assert`, as I seldom want the test to continue as it will just cause spurious errors:
```
	r, err := os.Open("testdata/metadata.golden")
	require.NoError(t, err)
￼ 	parsed, err := unmarshalTask(r)
	require.NoError(t, err)
￼ 	require.Equal(t, validMeta, *parsed)
```